base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -468,9 +471,124 @@ where
                 let output_statuses_by_tx_id = self.get_output_info_by_tx_id(tx_id)?;
                 Ok(OutputManagerResponse::OutputInfoByTxId(output_statuses_by_tx_id))
             },
+            OutputManagerRequest::CreateCommitmentProof(commitment, message, minimum_value) => self
+                .create_commitment_proof(commitment, message, minimum_value)
+                .await
+                .map(OutputManagerResponse::CommitmentProofCreated),
+            OutputManagerRequest::VerifyCommitmentProof(commitment, message, minimum_value, proof) => self
+                .verify_commitment_proof(commitment, message, minimum_value, proof)
+                .map(OutputManagerResponse::CommitmentProofVerified),
         }
     }
 
+    async fn create_commitment_proof(
```

Comment:
ship most of the work to the key manager. 
Its the only place where we should work with private keys.